ports/boost-iostreams/portfile.cmake

```
@@ -10,6 +10,9 @@ vcpkg_from_github(
 
 set(FEATURE_OPTIONS "")
 include("${CMAKE_CURRENT_LIST_DIR}/features.cmake")
+if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd_static")
```

```suggestion
    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd")
```
if it is necessary at all. This target is always created.

---

thanks, I changed to 
```
if (BOOST_IOSTREAMS_ENABLE_ZSTD)
    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd")
endif()
```

---

@dg0yt there's an error when `vcpkg install avro-cpp:x64-windows-static`:
```
CMake Error at D:/installed/x64-windows-static/share/boost_iostreams/boost_iostreams-targets.cmake:60 (set_target_properties):
  The link interface of target "Boost::iostreams" contains:

    zstd::libzstd

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  D:/installed/x64-windows-static/share/boost_iostreams/boost_iostreams-config.cmake:67 (include)
  D:/a/_work/1/s/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  D:/installed/x64-windows-static/share/boost/BoostConfig.cmake:67 (find_package)
  D:/installed/x64-windows-static/share/boost/BoostConfig.cmake:128 (boostcfg_find_component)
  D:/installed/x64-windows-static/share/boost/vcpkg-cmake-wrapper.cmake:3 (_find_package)
  D:/a/_work/1/s/scripts/buildsystems/vcpkg.cmake:813 (include)
  D:/downloads/tools/cmake-3.30.1-windows/cmake-3.30.1-windows-i386/share/cmake-3.30/Modules/FindBoost.cmake:610 (find_package)
  D:/installed/x64-windows-static/share/boost/vcpkg-cmake-wrapper.cmake:3 (_find_package)
  D:/a/_work/1/s/scripts/buildsystems/vcpkg.cmake:813 (include)
  CMakeLists.txt:77 (find_package)

```
[config-x64-windows-static-out.log](https://github.com/user-attachments/files/18220913/config-x64-windows-static-out.log)

but if I changed to 
```
if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd_static")
endif()
```
it works well, seems target zstd::libzstd not work when build static?
[boost_iostreams-targets.txt](https://github.com/user-attachments/files/18220944/boost_iostreams-targets.txt)
[zstdTargets.txt](https://github.com/user-attachments/files/18220945/zstdTargets.txt)